test/map/map.test.cpp

```diff
@@ -1390,11 +1372,19 @@ TEST(Map, DontKeepRenderData) {
         return res;
     };
     test.map.jumpTo(CameraOptions().withZoom(10));
-    test.map.getStyle().loadJSON(styleJSON);
-    test.frontend.render(test.map);
-    EXPECT_EQ(4, requestsCount);
-
-    test.map.getStyle().loadJSON(styleJSON);
-    test.frontend.render(test.map);
-    EXPECT_EQ(8, requestsCount);
+    const int iterations = 3;
+    // Keep render data.
+    for (int i = 1; i <= iterations; ++i) {
+        test.map.getStyle().loadJSON(styleJSON);
+        test.frontend.render(test.map);
+        EXPECT_EQ(4, requestsCount);
+    }
+    requestsCount = 0;
+    // Clear render data.
+    for (int i = 1; i <= iterations; ++i) {
+        test.frontend.getRenderer()->clearData();
+        test.map.getStyle().loadJSON(styleJSON);
+        test.frontend.render(test.map);
+        EXPECT_EQ(4 * i, requestsCount);
```

Comment:
Could this test be extended to assert what/whether other assets will be re-fetched (so beyond just tiles)?